### ISSUE TITLE
[Vanity URL] : Implementing a Vanity URL Resolver feature

### ISSUE THREAD

**Issue by jcastro-dotcms** [dotCMS/core#19173]
Found when working on Support: [102022](https://dotcms.zendesk.com/agent/tickets/102022)

**Is your feature request related to a problem? Please describe.**
When troubleshooting an issue from a customer related to Vanity URLs, it was necessary for me to identify the specific Vanity URL that was matching an incoming URL. However, there's currently no tool for that, so I had to create a JSP file that would take some parameters via Query String, and would display the information I needed.

**Describe the solution you'd like**
It'd be great to have a tool in the **Vanity URLs** portlet that allows users to test the Vanity URL that would handle an incoming request. This would be very useful for both the Enterprise Support Team and customers to test any of their Vanity URLs.

Depending on the RegEx used to redirect a request and a high number of Vanity URLs in the system, it can get very difficult to test or figure out the specific resolver for an incoming request. Also, the fact that users can determine the order in which Vanity URLs are evaluated can add more complexity to effectively testing the URL resolution.

**Describe alternatives you've considered**

**Additional context**
Help customers and the Enterprise Support Team troubleshoot conflicts and problems with Vanity URLs.

**Comment by wezell** [comment]
@jcastro-dotcms I think the vanity url that gets matched is now added to the response as a header?  See: #20416 . Does this solve at least the identification issue?

**Comment by jcastro-dotcms** [comment]
@wezell that is correct. Having an HTTP Header with the Vanity URL's Identifier solves most of the problem now. I think we can close this ticket.
